CONTRIBUTING.md

```diff
@@ -1,51 +1,150 @@
-   git fetch fork
-   git checkout -b <your-branch-name> fork/master
+   git remote add upstream https://github.com/ManimCommunity/manim.git
+   git pull upstream master
    ```
 
-   The first command tells `git` that you are going to make a Pull Request to Manim Community. 
-   The second command pulls all the commits from the aforementioned fork.
-   Finally, the third one makes your current working branch up-to-date with Manim Community's master branch.
-
-    Now there are three repositories that git is keeping track of: the manim community repo (referred to as "fork"), your own fork of it (referred to by git as "origin"), and your local repository.
+	After these commands, your local repository can keep track of your fork
+    (referred to as 'origin') as well as the main ManimCommunity repository
+    (referred to as 'upstream').
 
-4. After that, you can make your changes to the repo's files (the code is in the `manim` directory). Then, you can commit said changes.
+4. You can now make any changes to your local repository, add them, and commit
+   them.
 
-> As stated in the General Contribution Guidelines, make sure to test your changes against the example scenes in the `example_scenes` directory and test scenes in the `tests` directory, and add a new test in the `tests` directory for your new feature if you've made one.
-
-5. Finally, instead of  typing in `git push`, enter the command below.
+5. To make your changes visible in your GitHub fork, instead of typing in `git
+   push` as usual, you need to enter the command below.
 
    ```sh
    git push -u origin <your-branch-name>
    ```
 
-   Doing so creates a new branch with the updated contents of your fork on GitHub.
+   Doing so creates a new branch with the updated contents of your fork on
+   GitHub (the 'origin').
+
+6. If you want your changes to be incorporated to the main ManimCommunity
+   repository, you need to make sure that there are no merge conflicts between
+   the current upstream/master and the changes you are trying to make.  For
+   this purpose, please execute `git pull origin master` before you push
+   changes to your fork.  If this generates merge conflicts, make sure to
+   resolve them before submitting a PR.
+
+7. Further, make sure that you run the tests locally and that they all pass
+   before submitting a PR.  You can run the tests locally by executing `pytest`
+   from the main `manim/` folder (do not execute pytest from the code folder
+   `manim/manim`).
 
-   Then you can make a Pull Request to the Manim Community Repo from your fork, through GitHub. Make sure to select `ManimCommunity/manim` instead of `3b1b/manim` as the `base repository` and your fork and branch as `head repository` - see the picture below.
+8. Finally, make sure to use the `black` autoformatter before submitting a PR.
+   You can execute it by running `black <your_file>`.  This will make changes
+   to your code, and you will need to add/commit those changes.
```

Comment:
For me, this is a step that is a little annoying, primary because there is no good IDE support yet (I am using pycharm, maybe this is easier for other IDEs?) . Especially when you have more then one file, making this with every single file can get quit annoying. But maybe this is a point we can discuss in a separate issue. For now I am fine to leave it like this and get this pull request merged with this procedure.

Reply:
You can just run `black .` in the manim directory and it'll autoformat all files there, if they need to be changed; and given that the whole library has been preformatted, only the files you've changed will get formatted.

Reply:
Agree with @kilacoda . Running `black .` (on `manim/`) is enough.

Reply:
I agree that remembering to run black is annoying. But if you remember (for example, if it is explicitly described in the uigdelines! ;) ), running the command `black manim/` is fast and easy. There's literally no input necessary from the contributor/user.